### PR TITLE
skip non-existant properties instead of returning error

### PR DESCRIPTION
**Pull request by micah-white**
Relates to enterprise PR #1789

### CODE REVIEW

File: libtsuba/src/AddProperties.cpp [hunk 1]
```diff
@@ -108,14 +108,9 @@ tsuba::AddProperties(
...
-    if (!read_res) {
-      return read_res.error();
-    }
-    auto on_complete_res = on_complete(read_res.value());
-    if (!on_complete_res) {
-      return on_complete_res.error();
-    }
+    auto read_res = KATANA_CHECKED(future.get());
+
+    KATANA_CHECKED(on_complete(read_res));
```
Comment (tylershunt) [inline libtsuba/src/AddProperties.cpp:113]: nice! thanks for the cleanup.

File: libtsuba/src/RDGPartHeader.h [hunk 2]
```diff
@@ -432,9 +432,9 @@ class KATANA_EXPORT RDGPartHeader {
     for (const auto& name : names) {
       auto it = name_to_slot.find(name);
       if (it == name_to_slot.end()) {
-        return KATANA_ERROR(
-            ErrorCode::PropertyNotFound, "no property named {}",
-            std::quoted(name));
+        KATANA_LOG_DEBUG(
+            "Non-existant property \"{}\" requested. Skipping.", name);
+        continue;
```
Comment (tylershunt) [inline libtsuba/src/RDGPartHeader.h:437]: Sweet; Probably a matter of discussion but I would prefer `KATANA_LOG_WARN` here. I think this message would be helpful in customer engagements where we are trying to diagnose something and we pretty much never distribute DEBUG builds.

**Review by tylershunt — APPROVED** [review]
Seems like an easy approve. I guess when the user asks for non-existent properties this errors somewhere else now? (totally fine as an increment honestly) or does this just totally fix the issue?

File: libtsuba/src/RDGPartHeader.h (hunk 2 above)
Comment (micah-white) [reply libtsuba/src/RDGPartHeader.h:437]: Makes sense to me. I'll change it.

**Pull request merged**